beacon_node/http_api/src/lib.rs

```diff
@@ -657,87 +711,105 @@ pub fn serve<T: BeaconChainTypes>(
         .and(warp::path::end())
         .and_then(
             |state_id: StateId, chain: Arc<BeaconChain<T>>, query: api_types::CommitteesQuery| {
-                blocking_json_task(move || {
-                    state_id.map_state(&chain, |state| {
-                        let current_epoch = state.current_epoch();
-                        let epoch = query.epoch.unwrap_or(current_epoch);
-
-                        let committee_cache = match RelativeEpoch::from_epoch(current_epoch, epoch)
-                        {
-                            Ok(relative_epoch)
-                                if state.committee_cache_is_initialized(relative_epoch) =>
-                            {
-                                state.committee_cache(relative_epoch).map(Cow::Borrowed)
-                            }
-                            _ => CommitteeCache::initialized(state, epoch, &chain.spec)
-                                .map(Cow::Owned),
-                        }
-                        .map_err(|e| match e {
-                            BeaconStateError::EpochOutOfBounds => {
-                                let max_sprp = T::EthSpec::slots_per_historical_root() as u64;
-                                let first_subsequent_restore_point_slot =
-                                    ((epoch.start_slot(T::EthSpec::slots_per_epoch()) / max_sprp)
-                                        + 1)
-                                        * max_sprp;
-                                if epoch < current_epoch {
-                                    warp_utils::reject::custom_bad_request(format!(
-                                        "epoch out of bounds, try state at slot {}",
-                                        first_subsequent_restore_point_slot,
-                                    ))
-                                } else {
-                                    warp_utils::reject::custom_bad_request(
-                                        "epoch out of bounds, too far in future".into(),
-                                    )
-                                }
-                            }
-                            _ => warp_utils::reject::beacon_chain_error(e.into()),
-                        })?;
+                // The api spec says if the epoch is not present in the request then the epoch
+                // of the state should be used.
+                let query_state_id = query.epoch.map_or(state_id, |epoch| {
+                    StateId::from_slot(epoch.start_slot(T::EthSpec::slots_per_epoch()))
+                });
```

Comment:
These lines probably made it back in during a merge conflict, but they'll cause a performance regression by undoing the logic from https://github.com/sigp/lighthouse/pull/3272.

We should always use the `state_id` provided with the call.

Thanks @jclapis for the find

Reply:
:pray:

Reply:
I've done a hotfix on the `builder-api-rocket-pool` branch, which could be cherry-picked here if you'd like Mac https://github.com/sigp/lighthouse/commit/0edf7cc9f46a887614ab8743ef30bf941a3b54f6

Reply:
Good catch! Sorry for any inconveniences :grimacing: